### ISSUE TITLE
how to access webinterface with ipv6

### ISSUE THREAD

**Issue by produnis** [pyload/pyload#2795]
Hi there,
I have a IPv6 network, and I wonder what to do in order to make pyLoad webinterface listen on IPv6?

**Comment by carazzim0** [comment]
I'd suggest you put a Reverse-Proxy infront of pyLoad that listens on both IPv4 and IPv6 and proxies all requests to pyLoad ;)

**Comment by produnis** [comment]
Yeah, currently I do so using 6tunnel.

But as it is 2017, and IPv6 is about 20years old, I thought I ask for native support.

**Comment by treysis** [comment]
I can only second that request. I'd love to see native IPv6 support. And not only for the web interface, but also for downloading from IPv6-enabled OCH (like openload for example).

**Comment by treysis** [comment]
Still an issue!

**Comment by treysis** [comment]
Well, you can manually enter specific IPv6 addresses, but you can't use "::" to listen on all addresses. Also, you can't run DualStack. Couldn't figure out where this is parsed from the config.

**Comment by treysis** [comment]
Ok, I dug a little bit deeper. The problem is that even though the (threaded) wsgiserver (CherryPy) supports DualStack with listening on all addresses, it needs `::` as host listen address to do this. Unfortunately, setting the listening address to `::` in pyload.conf confuses `module/ConfigParser.py` due to the following "sanity check":

``` (line 149-153)
elif line.endswith(":"):
    section, none, desc = line[:-1].partition('-')
    section = section.strip()
    desc = desc.replace('"', "").strip()
    conf[section] = {"desc": desc}
```

I haven't found a good way to improve the parsing algorithm. Instead, I decided to make pyLoad treat "0.0.0.0" as "::" and thus enabling the server to run dual-stacked. I did this by editing `module/web/ServerThreaded.py` and changed line 24 like this:

```
self.host = pycore.config['webinterface']['host']
    if self.host == "0.0.0.0":
    self.host = "::"
self.port = pycore.config['webinterface']['port']
```

Now you can't run it only on 0.0.0.0 (INADDR_ANY) alone, but it will both be running on INADDR_ANY and IN6ADDR_ANY. However, I guess if somebody is so stupid to completely disable IPv6 via kernel parameter, this would cause pyLoad webinterface to fail to start up (I tried to add another option "dualstack" = "True/False" to pyload.conf, but when trying to parse this value, ConfigParser.py throws an error for whatever reason I fail to comprehend).

**Comment by treysis** [comment]
Closed by #3619
